### ISSUE TITLE
JogWheelBasic component doesn't take into account deck switching

### ISSUE THREAD

**Issue by miwig** [mixxxdj/mixxx#11867]
### Bug Description

As seen [here](https://github.com/mixxxdj/mixxx/blob/dfde28dbe44375706abcf8846e7fd099c1c66a87/res/controllers/midi-components-0.0.js#L772) the `JogWheelBasic` component doesn't take into account the selected deck/group when scratching.

I could try a PR if you want. I assume the fix would be to add special handling [here](https://github.com/mixxxdj/mixxx/blob/dfde28dbe44375706abcf8846e7fd099c1c66a87/res/controllers/midi-components-0.0.js#L707) similar to how it's done for `EffectAssignmentButton`.

### Version

2.3.5

### OS

Arch Linux btw

**Comment by Swiftb0y** [comment]
Hey there. Thank you for filing the issue. This is indeed an oversight. I'd be very happy if you could file a PR for that. I see two possible solutions:
1. The special handling as you suggested
2. instead of relying on the `.deck`, we instead extract the deck number from the `.group` on each call. 

The second solution is more robust, but likely very much lower-performant... Not sure what solution is the most optimal one. What do you think?

**Comment by miwig** [comment]
> instead of relying on the .deck, we instead extract the deck number from the .group on each call.

Couldn't `JogWheelBasic` override `connect` to extract the deck number from the group?

**Comment by Swiftb0y** [comment]
That... could work... Not sure how robust that is though.

Another option would to do the regex extraction on each call, but speed it up by memoizing the call each time.

**Comment by miwig** [comment]
> That... could work... Not sure how robust that is though.

It seems more robust than the special case solution to me. The docs also suggest that overriding `connect` is allowed, but give a different reason for doing it.

**Comment by Swiftb0y** [comment]
Right, overwriting `connect` is completely reasonable and encouraged. I'm just not 100% confident that consumers will always call `connect` after changing the `group`. On the other hand, they should, otherwise that would probably result in bugs in all other components as well. So yeah, overwriting `connect` and extracting the `deck` from the `group` is probably the best solution. Can you open a PR with that?

**Comment by miwig** [comment]
I can probably get around to it this week, yeah.

**Comment by SamWhited** [comment]
Just ran into this today and found this very old issue; it appears that a solution was merged and this issue can be closed? (although maybe someone else wants to verify this first? As far as I can tell it's still not working for me despite connect being called, but I can't see how that would be the case)

**Comment by Swiftb0y** [comment]
Yup. Thanks for the reminder.
